### DISCUSSION TITLE
[Idea] Doctrine Entity - Table Naming Strategy for Modules

### DISCUSSION THREAD

**Discussion by Kaikina** [PrestaShop/PrestaShop#36690] [Ideas]
### The problem

I usually prefix my modules tables with a module prefix (usually the module name). It makes clear the identification of tables related to a functionnality.
Otherwise they get lost in a huge amount of dbprefix_tables

Problem, it forces me to name my Doctrine Entities with this module prefix : MyPrefixEntity instead of Entity.
Thus, all my module entities start with the same name, for no valid reason.

### The solution

In the DoctrineNamingStrategy class, in the classToTableName function, we could check if the className (which includes its whole namespace) includes "Module". If yes, it means we are loading a module entity class.
If that so, instead of returning $this->prefix . {snake_case entity name}, we could implement a new $underscored variable which would be the module name in snake case.

It is easy to parse the namespace to retrieve the name that is following the "Module" part : 
PrestaShop\Module\MyModule\Entity\MyEntity.php => my_module

Then instead of returning for instance : ps_my_entity
we would return : ps_my_module_my_entity

While this fits my own use case and my own naming strategy, I don't know if 
- This is good naming practice
- This would not be too restrictive for other developers

### Alternatives

I don't know where the change would take place, but in the Entity annotation, we can name the table related to the entity, PrestaShop would have to retrieve this name if defined, and simply include the dbPrefix to it.

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by PululuK** [comment]
Hello @Kaikina 

You can get around this as follows:

```php

<?php

namespace PrestaShop\Module\MyModule\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Table(name=PrestaShop\Module\MyModule\Entity\MyEntity::MY_TAB_NAME)
 */
class MyEntity
{
    public const MY_TAB_NAME = _DB_PREFIX_.'my_module_custom_tab_name';
}

**Reply by Kaikina** [reply]
Oh !! I tried inserting a variable in the annotation but didn't succeed. I like this way ! Thanks !
